beacon-chain/sync/rpc_data_column_sidecars_by_root.go

```diff
@@ -133,50 +134,66 @@ func (s *Service) dataColumnSidecarByRootRPCHandler(ctx context.Context, msg int
 		// TODO: Differentiate between blobs and columns for our storage engine
 		// If the data column is nil, it means it is not yet available in the db.
 		// We wait for it to be available.
-		// TODO: Use a real feed like `nc := s.blobNotifiers.forRoot(root)` instead of this for/sleep loop looking in the DB.
-		var sc *eth.DataColumnSidecar
 
-		for {
-			sc, err = s.cfg.blobStorage.GetColumn(root, idx)
-			if err != nil {
-				if ctxErr := ctx.Err(); ctxErr != nil {
-					closeStream(stream, log)
-					return ctxErr
-				}
+		// Subscribe to the data column channel for this root.
+		rootChannel := s.cfg.blobStorage.DataColumnNotifier.ForRoot(requestedRoot)
+		defer s.cfg.blobStorage.DataColumnNotifier.Delete(requestedRoot)
```

Comment:
If you have multiple requests which are waiting for it, is it safe to delete the notifier before all requests have been fulfilled ?

Reply:
Good point. Will try to use [event feed.](https://github.com/prysmaticlabs/prysm/blob/d20d0fa738c54d7de0656ed82bbf91766f5708a9/async/event/feed.go). It seems the right tool to use in this case.

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/14062/commits/7d25e087f1cc8a72a5193dfe7e7e671341dee8c3.